### PR TITLE
Fix getting the certificate bundle for dav external storage

### PR DESCRIPTION
**Pull request by icewind1991**
Required in (some?) cases to get fed. sharing working with self-signed certificates

cc @owncloud/filesystem

### CODE REVIEW

File: lib/private/Files/Storage/DAV.php [hunk 1]
```diff
@@ -812,6 +816,7 @@ public function hasUpdated($path, $time) {
 	 * which might be temporary
 	 */
 	private function convertException(Exception $e, $path = '') {
+		\OC::$server->getLogger()->logException($e);
```
Comment (PVince81) [inline lib/private/Files/Storage/DAV.php]: hmm, I thought we logged the exception already later

**Comment by PVince81** [comment]
Code looks good :+1:

File: lib/private/Files/Storage/DAV.php (hunk 1 above)
Comment (icewind1991) [inline lib/private/Files/Storage/DAV.php]: From what I can tell we log the converted exception in some cases.

Having the original exception makes it easier to track down the cause

File: lib/private/Files/Storage/DAV.php [hunk 2]
```diff
@@ -107,7 +107,11 @@ public function __construct($params) {
 			}
 			if ($this->secure === true) {
 				// inject mock for testing
-				$certPath = \OC_User::getHome(\OC_User::getUser()) . '/files_external/rootcerts.crt';
```
Comment (DeepDiver1975) [inline lib/private/Files/Storage/DAV.php]: are there any migration steps required to move this file around?

Comment (icewind1991) [inline lib/private/Files/Storage/DAV.php]: the path returned should be the same.

I think the issue I ran into was the cert bundle not yet being created for the user, using the cert manager makes sure things like that are all taken care off

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**